blosc/shuffle-sse2.c

```diff
@@ -409,35 +439,56 @@ shuffle_sse2(const size_t bytesoftype, const size_t blocksize,
 void
 unshuffle_sse2(const size_t bytesoftype, const size_t blocksize,
                const uint8_t* const _src, uint8_t* const _dest) {
-  int unaligned_src = (int)((uintptr_t)_src % 16);
-  int unaligned_dest = (int)((uintptr_t)_dest % 16);
-  int multiple_of_block = (blocksize % (16 * bytesoftype)) == 0;
-  int too_small = (blocksize < 256);
-
-  if (unaligned_src || unaligned_dest || !multiple_of_block || too_small) {
-    /* _src or _dest buffer is not aligned, not multiple of the vectorization
-     * size or is not too small.  Call the generic routine. */
+  const size_t vectorized_chunk_size = bytesoftype * sizeof(__m128i);
+  
+  int unaligned_src = (int)((uintptr_t)_src % sizeof(__m128i));
+  int unaligned_dest = (int)((uintptr_t)_dest % sizeof(__m128i));
+  int too_small = (blocksize < vectorized_chunk_size);
+
+  if (unaligned_src || unaligned_dest || too_small) {
```

Comment:
Provided that all the stores are made through `storeu`, maybe we can remove the 'unaligned_dest' condition here?